src/Range.jsx

```diff
@@ -260,23 +260,25 @@ class Range extends React.Component {
     return true;
   }
 
-  trimAlignValue(v, nextProps = {}) {
+  trimAlignValue(v, i, nextProps = {}) {
     const mergedProps = { ...this.props, ...nextProps };
     const valInRange = utils.ensureValueInRange(v, mergedProps);
-    const valNotConflict = this.ensureValueNotConflict(valInRange, mergedProps);
+    const valNotConflict = this.ensureValueNotConflict(i, valInRange, mergedProps);
     return utils.ensureValuePrecision(valNotConflict, mergedProps);
   }
 
-  ensureValueNotConflict(val, { allowCross }) {
+  ensureValueNotConflict(i, val, { allowCross, pushable: thershold }) {
     const state = this.state || {};
-    const { handle, bounds } = state;
+    const { bounds } = state;
+    const handle = (i === undefined) ? state.handle : i;
+    thershold = Number(thershold);
     /* eslint-disable eqeqeq */
-    if (!allowCross && handle != null) {
-      if (handle > 0 && val <= bounds[handle - 1]) {
-        return bounds[handle - 1];
+    if (!allowCross && handle != null && bounds !== undefined) {
+      if (handle > 0 && val <= (bounds[handle - 1] + thershold)) {
```

Comment:
Actually, handle is a array index, so it should always greater than 0

Reply:
No, handle will be `null` only one moment when init component, `constructor` init state `this.state = {
      handle: null,
      recent,
      bounds,
    };`

once click any handle,  handle will be no more `null`.

Reply:
oops, sorry, I am mistaken about you.
Here should be judge safe distance of the handle and surrounding handles.
handle is index, `handle <= bounds.length-1 && handle >= 0`, with equal.